zip-0213.rst

```diff
@@ -0,0 +1,179 @@
+[#zip-0200]_.
+
+The term "Sapling" in this document is to be interpreted as described in ZIP 205
+[#zip-0205]_.
+
+The terms "Founders' Reward", "funding stream", and "active funding stream" in this
+document are to be interpreted as described in ZIP 207 [#zip-0207]_.
+
+Abstract
+========
+
+This proposal defines modifications to the Zcash consensus rules that enable coinbase
+funds to be mined to Sapling addresses. It does not disable the use of transparent
+addresses in coinbase transactions.
+
+
+Motivation
+==========
+
+Zcash inherited the concept of "coinbase transactions" from Bitcoin: special transactions
+inside each block that are allowed to have no inputs. These transactions are created by
+miners during block creation, and collect the block reward and transaction fees into new
+transparent outputs that can then be spent. They are also leveraged in Zcash for the
+Founders' Reward (and potentially for funding streams [#zip-0207]_).
+
+On the path to deprecating and removing Bitcoin-inherited transparent addresses within the
+Zcash network, a required step is to be able to create coinbase transactions that have no
+transparent outputs. However, Zcash was launched with a consensus rule preventing coinbase
+transactions from containing shielded outputs, instead enforcing that coinbase funds could
+not be spent in transactions with transparent outputs. This was partly in order to reduce
+the complexity of the original Zcash modifications to the Bitcoin Core codebase, but also
+because at the time, shielded transactions required significant memory and CPU resources
+to create.
+
+The Sapling network upgrade [#zip-0205]_ deployed architectural changes and performance
+improvements that make shielding funds directly in the coinbase transaction feasible. In
+order to reduce the complexity of the Sapling network upgrade, the existing consensus
+rules preventing coinbase transactions from containing shielded outputs were extended to
+cover Sapling outputs. Therefore, it is now necessary to modify the consensus rules in
+order to enable miners to start using Sapling addresses.
+
+
+Specification
+=============
+
+Prior to activation of the TODO network upgrade, the existing consensus rule requiring
+coinbase transactions to have zero Sapling outputs is enforced.
+
+Once the TODO network upgrade activates:
+
+- The above existing consensus rule is no longer active, and coinbase transactions MAY
+  contain Sapling outputs.
+
+- The consensus rules applied to ``valueBalance``, ``vShieldedOutput``, and ``bindingSig``
+  in non-coinbase transactions MUST also be applied to coinbase transactions.
+
+- The existing consensus rule requiring transactions that spend coinbase outputs to have
+  an empty ``vout``, is amended to only apply to transactions that spend transparent
+  coinbase outputs.
+
+- The existing consensus rule requiring coinbase outputs to have 100 confirmations before
```

Comment:
Ditto here; coinbase maturity rules have never affected shielded outputs (because they couldn't), so we should merely explicitly say this here rather than suggesting a rule change.

Reply:
Here, the protocol specification already handles this:

> A transaction MUSTNOT spend an output of a coinbase transaction (necessarily a transparent output) from a block less than 100 blocks prior to the spend. Note that outputs ofcoinbase transactions include Founders’ Reward outputs.

And the `zcashd` consensus rules I agree don't require changing. So we could maybe adjust this. OTOH, `zcashd` _does_ need altering in this regard, because the wallet code assumes a blanket 100-block coinbase transaction maturity.